### DISCUSSION TITLE
Error while login with Google provider. User is `null`

### DISCUSSION THREAD

**Discussion by mscalessio** [nuxt-community/auth-module#1502] [Support Requests]
### Version
module: "5.0.0-1617968180.f699074"
nuxt: "^2.15.3"

#### [mode](https://nuxtjs.org/api/configuration-mode): 
 - [x] universal
 - [ ] spa

### Nuxt configuration
```
auth: {
    strategies: {
      google: {
        clientId: ***********-**********************.apps.googleusercontent.com,
        scope: ['profile', 'email'],
        codeChallengeMethod: '',
        responseType: 'token id_token',
        redirectUri: 'http://localhost:3000/dashboard`,
      },
    },
}
```

#### What is expected?
`user` object in the nuxt store filled with the user information fetched from google endpoint and `loggedIn: true`! 

#### What is actually happening?
`user` object in nuxt store is `null`, `loggedIn: false` but in cookie and in localstorage key `auth.google.state` is filled with the one from google callback

#### Steps to reproduce
Simply try to login with `this.$auth.loginWith('google')`

## Checklist
* [x]  I have tested with the latest Nuxt version and the issue still occurs
* [x]  I have tested with the latest module version and the issue still occurs
* [x]  I have searched the issue tracker and this issue hasn't been reported yet

### What is actually happening?
The callback url from google after selecting  my account is something like that:
```
http://localhost:3000/dashboard#state=7kSmuKe1d3&access_token=ya29.a0AfH6SM
...LoNEqm3Zfibp&token_type=Bearer&expires_in=3599&scope=email%20profile%20https:/www.googleapis.com/auth/userinfo.profile%20https:/www.googleapis.com/auth/userinfo.email%20openid&id_token=eyJhbGciOiJSUz
...asdASD2&authuser=0&prompt=consent
```
but the store remain in this state
![nuxt-auth-google](https://user-images.githubusercontent.com/4687670/114790546-37f5d100-9d85-11eb-94a3-231a8b147cf2.png)

Someone have the same issue?
Thank you in advice.

**Comment by GuasaPlay** [comment]
@mscalessio I'm using the same version of @ nuxtjs / auth-next but it worked for me.

My config
```
google: {
        clientId: 'xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx.apps.googleusercontent.com',
        codeChallengeMethod: '',
        scope: ['profile', 'email'],
        responseType: 'token id_token'
      }
```
My package.json
```
"dependencies": {
    "@nuxtjs/auth-next": "^5.0.0-1617968180.f699074",
    "@nuxtjs/axios": "^5.13.1",
    "@nuxtjs/dayjs": "^1.3.1",
    "@nuxtjs/pwa": "^3.3.5",
    "core-js": "^3.10.1",
    "nuxt": "^2.15.4"
  },
```

![image](https://user-images.githubusercontent.com/45466316/114791483-3fe54f00-9d4c-11eb-8d39-a7632d112da6.png)

**Comment by jonalxh** [comment]
Same here.

**Comment by JoseJuan81** [comment]
I'm getting the same issue, my user object is empty and `loggedIn: false`

**Comment by dajpes** [comment]
@jonalxh @JoseJuan81 The same happened to me until I tried @GuasaPlay 's solution. What if you try what I mentioned [here](https://github.com/nuxt-community/auth-module/issues/1199)

Completely removing from the url code_challenge_method and nounce parameters

**Comment by anhthang** [comment]
I'm having same issue, even I tested with Discord provider.

**Comment by subramanya92** [comment]
@mscalessio I had the same issue, I think the problem occurs if you have `login` and `home` parameters same. Try changing it to a different one like below, it worked for me. 
```
auth {
  redirect: {
      login : '/'
      callback: '/',
      home: '/home',  
    }
}
```
After the fix it was obvious why it was happening :)

**Comment by timbog** [comment]
Hi @mscalessio , facing the same issue, have you found the way how to fix it?

**Comment by pavanasrivinaya** [comment]
Yes facing the same issue ..Can anyone fix it

**Comment by prikr** [comment]
Is this bug already fixed? Getting the same result.

**Comment by bmulholland** [comment]
`user` is fetched from the user info endpoint. Make sure that's working and returning a valid hash.

**Reply by papayaah** [reply]
Hi @bmulholland how do we test this?

**Reply by bmulholland** [reply]
https://auth.nuxtjs.org/schemes/local/#fetch-user
